allensdk/brain_observatory/ecephys/current_source_density/_interpolation_utils.py

```diff
@@ -0,0 +1,253 @@
+
+    return actual_channel_locations[min_chan:, :]
+
+
+def make_interp_channel_locations(min_chan: int = 0,
+                                  max_chan: int = 384) -> np.ndarray:
+    '''Generate x/y locations for interpolated Neuropixels recording sites.
+
+    This version just returns the central column of interpolated sites.
+
+          0  8 16 24 32 40 48
+     0    -  -  *  o  -  -  * <-- actual recording site (*)
+    10    -  -  -  o  -  -  -
+    20    *  -  -  o  *  -  -
+    30    -  -  -  o  -  -  -
+    40    -  -  *  o  -  -  *
+    50    -  -  -  o  -  -  -
+    60    *  -  -  o  *  -  -
+                   ^
+                   interpolated column sites (o)
+
+    Parameters
+    ----------
+    min_chan : int, optional
+        Lowest channel number to use, by default 0
+    max_chan : int, optional
+        Highest channel number to use, by default 384
+
+    Returns
+    -------
+    interp_channel_locations: numpy.ndarray
+        column 1 = interpolated x positions in microns
+        column 2 = y positions in microns
+    '''
+
+    # interp_channel_locations = np.zeros(((max_chan-min_chan)*7,2))
+    # xlocations = [0, 8, 16, 24, 32, 40, 48]
+
+    # for i in range(min_chan,(max_chan-min_chan)*7+min_chan):
+    #     interp_channel_locations[i,0] = xlocations[i%7]
+    #     interp_channel_locations[i,1] = np.floor(i/7)*10
+
+    # in_central_column = np.where(interp_channel_locations[:,0] == 24)[0]
+
+    # interp_channel_locations = interp_channel_locations[in_central_column]
+
+    # return interp_channel_locations
+
+    interp_channel_locations = np.zeros((max_chan, 2))
+
+    for ch in range(min_chan, max_chan):
+        interp_channel_locations[ch, 0] = 24
+        interp_channel_locations[ch, 1] = ch * 10
+
+    return interp_channel_locations[min_chan:, :]
+
+
+def interp_channel_locs(trial_lfp: np.ndarray,
+                        indices: np.ndarray,
+                        num_channels: int,
+                        method: str = 'cubic') -> Tuple[np.ndarray, float]:
+    '''Interpolates single-trial lfp channel locations to account for
+    channel stagger.
+
+    Parameters
+    ----------
+    trial_lfp : numpy.ndarray
+        Single trial lfp data in the form of: channels x time samples
+    indices : numpy.ndarray
+        Indicies of "good" channels (neither reference nor noisy)
+    num_channels : int
+        Number of channels in original data
+    method : str, optional
+        Interpolation method ['cubic', 'linear', 'nearest'], by default 'cubic'
+
+    Returns
+    -------
+    Tuple[interp_data, spacing]
+        interp_lfp: numpy.ndarray
+            Interpolated single-trial lfp data in the form of:
+            channels x time samples
+        spacing: float
+            Distance between interpolated sites (in millimeters)
+    '''
+
+    num_samples = trial_lfp.shape[1]
+
+    actual_locs = make_actual_channel_locations(0, num_channels)
```

Comment:
I would rather see `actual_locs` and `interp_locs` as parameters to the function.

Reply:
I've moved `make_actual_channel_locations()` and `make_interp_channel_locations()` out of this function.